Sources/CardPayments/APIRequests/ConfirmPaymentSourceRequest.swift

```diff
@@ -35,13 +36,52 @@ struct ConfirmPaymentSourceRequest: APIRequest {
         
         path = String(format: pathFormat, orderID)
         
-        jsonEncoder.keyEncodingStrategy = .convertToSnakeCase
-        body = try jsonEncoder.encode(confirmPaymentSource)
+        jsonEncoder?.keyEncodingStrategy = .convertToSnakeCase
+        do {
+            body = try jsonEncoder?.encode(confirmPaymentSource)
+        } catch {
+            throw CardClientError.encodingError
+        }
         
         // TODO - The complexity in this `init` signals to reconsider our use/design of the `APIRequest` protocol.
         // Existing pattern doesn't provide clear, testable interface for encoding JSON POST bodies.
     }
     
+    /// For testing purpose
+    init(
+        accessToken: String,
+        cardRequest: CardRequest,
+        encoder: JSONEncoder?
+    ) throws {
```

Comment:
I am not sure how we would do this since the other initializer instantiates JSONEncoder.
I can't just call the original initializer from the test initializer. 

Originally, I had single initializer with optional encoder parameter and toggled assigning jsonEncoder property depending on whether encoder parameter was provided.

Reply:
We do this extensively throughout the SDK, injecting mocks via dependency injection. [Here is an example](https://github.com/paypal/iOS-SDK/blob/2d2b08f14ac487da68cb1f1278cfac3c17c45dc6/Sources/CorePayments/Networking/HTTP.swift#L11-L15) using default params. [Here is an example](https://github.com/paypal/iOS-SDK/blob/2d2b08f14ac487da68cb1f1278cfac3c17c45dc6/Sources/CardPayments/CardClient.swift#L16-L29) exposing 2 inits.

Reply:
Fixed in this commit. be9e5acdfbd4a8c9209de3cd5ba073f4f76ff494 and deleted "for testing purposes" comment in commit.

Reply:
Thank you Sammy.